lib/pos_network.go

```diff
@@ -365,15 +364,21 @@ func (sig *AggregatedBLSSignature) Eq(other *AggregatedBLSSignature) bool {
 		return false
 	}
 
-	return bytes.Equal(sig.SignersList, other.SignersList) &&
-		sig.Signature.Eq(other.Signature)
+	if !sig.Signature.Eq(other.Signature) {
+		return false
+	}
+
+	return bytes.Equal(sig.SignersList.ToBytes(), other.SignersList.ToBytes())
 }
 
 func (sig *AggregatedBLSSignature) ToBytes() ([]byte, error) {
 	retBytes := []byte{}
 
 	// SignersList
-	retBytes = append(retBytes, EncodeByteArray(sig.SignersList)...)
+	if sig.SignersList == nil {
+		return nil, errors.New("AggregatedBLSSignature.ToBytes: SignersList must not be nil")
+	}
+	retBytes = append(retBytes, EncodeByteArray(sig.SignersList.ToBytes())...)
```

Comment:
I had a related comment for #564, which would remove the need for `EncodeByteArray`, but this route is fine too.

Reply:
Good point. I do think that the `lib` package should be responsible for encoding the bitlist for network messages, but you're right that using the raw EncodeByteArray here is non-ideal.

I added a new `EncodeBitset` and `DecodeBitset` functions to wrap the byte encoding and do the length prefixing.